### PR TITLE
Add abs(__int64) overload for 64-bit targets. Fixes #143

### LINKED ISSUE #143: abs overload for 64-bit ints missing when building with clang 3.5 (?)
When building with clang I get the following warning and variations thereof:
```
In file included from /home/poizan/coreclr/src/jit/jitpch.h:20:
/home/poizan/coreclr/src/jit/jit.h:608:31: warning: absolute value function 'abs' given an argument of type 'ssize_t' (aka 'long') but has parameter of type 'int' which may cause truncation of value [-Wabsolute-value]
    return ((size_t)          abs(x));
                              ^
/home/poizan/coreclr/src/jit/jit.h:608:31: note: use function 'std::abs' instead
    return ((size_t)          abs(x));
                              ^~~
                              std::abs
/home/poizan/coreclr/src/jit/jit.h:608:31: note: include the header <cstdlib> or explicitly provide a declaration for 'std::abs'
```

This seems a bit weird as jitpch.h does in fact include cstdlib. But it also includes stdlib.h before that, so maybe the clang version of cstdlib fails to take this possibility into account.

clang version:
```
Debian clang version 3.5.0-9 (tags/RELEASE_350/final) (based on LLVM 3.5.0)
Target: x86_64-pc-linux-gnu
Thread model: posix
```

### PR DESCRIPTION
**Pull request by poizan42**
#143

### CODE REVIEW

File: src/pal/inc/pal.h [hunk 1]
```diff
@@ -5466,7 +5466,14 @@ PALIMPORT double __cdecl _copysign(double, double);
 extern "C++" {
 inline float fabsf(float _X)
 {
-    return ((float)fabs((double)_X)); }
+    return ((float)fabs((double)_X));
+}
+
+#ifdef BIT64
+inline __int64 abs(__int64 _X) {
+    return labs(_X);
```
Comment (jkotas) [inline src/pal/inc/pal.h]: labs in PAL takes LONG argument. LONG is 32-bit type. This code seems to be losing the upper 32-bits.

Comment (poizan42) [inline src/pal/inc/pal.h]: Oh, right, but then we are completely missing an implementation of abs(__int64). Should we call the C version llabs then?

Comment (jkotas) [inline src/pal/inc/pal.h]: Yes. It may be also nice to add a small test for it to make sure that it works.

**Comment by sergiy-k** [comment]
@poizan42  LGTM. And thank you so much for adding a test!

**Comment by jkotas** [comment]
LGTM

**Pull request merged**